lib/bundler/source/git/git_proxy.rb

```diff
@@ -235,26 +238,15 @@ def allowed_in_path
           raise GitError, "The git source #{uri} is not yet checked out. Please run `bundle install` before trying to start your application"
         end
 
-        # TODO: Replace this with Open3 when upgrading to bundler 2
-        # Similar to #git_null, as Open3 is not cross-platform,
-        # a temporary way is to use Tempfile to capture the stderr.
-        # When replacing this using Open3, make sure git_null is
-        # also replaced by Open3, so stdout and stderr all got handled properly.
-        def capture_and_filter_stderr(uri)
-          return_value, captured_err = ""
-          backup_stderr = STDERR.dup
-          begin
-            Tempfile.open("captured_stderr") do |f|
-              STDERR.reopen(f)
-              return_value = yield
-              f.rewind
-              captured_err = f.read
-            end
-          ensure
-            STDERR.reopen backup_stderr
-          end
+        def capture_and_filter_stderr(uri, cmd)
```

Comment:
Should we define argument order same as `capture_and_ignore_stderr` like ```capture_and_filter_stderr(cmd, uri)```?

I prefer to same arguments order both. But It\'s not strong opinion.

Reply:
But `capture_and_ignore_stderr` receives no `uri`, so there's no really a "defined order". `cmd` is the first argument of that method, but it's also the last argument :)

I have no strong opinion either, I only defined it like this so that it was more "visually similar" to the old way:

```diff
-            capture_and_filter_stderr(uri) { `git #{command}` }
+            capture_and_filter_stderr(uri, "git #{command}")
```

Reply:
These are all private utilities, so we can always change them in the future if we find a different way is more readable. So I'll merge this! Thanks for having a look @hsbt!!